### PR TITLE
Move roslyn analyzers over

### PR DESCRIPTION
**Pull request by genlu**
- Copy Roslyn.Diagnostics.Analyzers projects over from dotnet/roslyn repo.
- Update target files and NuGet package references.
- Add projects to solution and hook up tests execution.
- Add vxis installer project for Roslyn.Diagnostics.Analyzers.
- Fix a string formatting bug in Microsoft.CodeAnalysis.UnitTests.DiagnosticAnalyzerTestBase.GetGlobalResult.
- Add NuGet Authoring project to Roslyn.Diagnostic.Analyzers. The name and structure of the NuGet package remains identical as before:
    1. The name is Microsoft.Net.RoslynDiagnostics.
    2. Unlike other analyzer projects in this repo, this one doesn't have install/uninstall scripts and has a props file instead.

@tmeschter @jaredpar @srivatsn @mavasani

### CODE REVIEW

File: src/Roslyn/Core/Microsoft.Net.RoslynDiagnostics.nuspec [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  <metadata>
+    <id>Microsoft.Net.RoslynDiagnostics</id>
+    <description>
+      Roslyn Diagnostics Toolset package. Contains analyzers specific to the .NET Compiler Platform ("Roslyn") project.
+
+      Supported Platforms:
+      - .NET Framework 4.5
+    </description>
+    <dependencies>
+      <dependency id="Microsoft.Net.ToolsetCompilers" version="$currentVersion$" />
```
Comment (genlu) [inline src/Roslyn/Core/Microsoft.Net.RoslynDiagnostics.nuspec]: @tmeschter @jaredpar  Will these two package versions always be in sync or we need to manually assign a fixed version here?

File: build/Targets/Analyzers.Versions.targets [hunk 2]
```diff
@@ -17,11 +17,17 @@
     
     <SystemRuntimeInteropServicesAnalyzersSemanticVersion>1.0.2</SystemRuntimeInteropServicesAnalyzersSemanticVersion>
     <SystemRuntimeInteropServicesAnalyzersPreReleaseVersion>beta1</SystemRuntimeInteropServicesAnalyzersPreReleaseVersion>
+    
+    <SystemSecurityCryptographyHashingAlgorithmsAnalyzersSemanticVersion>1.0.0</SystemSecurityCryptographyHashingAlgorithmsAnalyzersSemanticVersion>
+    <SystemSecurityCryptographyHashingAlgorithmsAnalyzersPreReleaseVersion>beta1</SystemSecurityCryptographyHashingAlgorithmsAnalyzersPreReleaseVersion>
```
Comment (genlu) [inline build/Targets/Analyzers.Versions.targets]: Not sure how this get in this branch, it belongs to another branch of mine. Will remove when addressing CR comments.

File: src/Roslyn/Core/Microsoft.Net.RoslynDiagnostics.nuspec (hunk 1 above)
Comment (tmeschter) [inline src/Roslyn/Core/Microsoft.Net.RoslynDiagnostics.nuspec]: @genlu The `<dependency>` element should be removed entirely. While the analyzer APIs were in flux it made sense to keep these packages in sync, but now that Roslyn 1.0.0 has RTM'd it's not needed any more.

**Comment by tmeschter** [comment]
LGTM.

**Comment by jaredpar** [comment]
LGTM

**Comment by srivatsn** [comment]
:+1:

**Pull request merged**